### PR TITLE
Add GenericType to the signature of ServerMessageBodyWriter methods

### PR DESCRIPTION
**Pull request by geoand**
This is done in a way that doesn't make us look up the types using generics

The instigator for this was that Qson needs this generic type

### CODE REVIEW

**Comment by geoand** [comment]
I added a second commit this enhances the generic type handling when a resource method return type is a type variable.

File: independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java [hunk 1]
```diff
@@ -609,29 +618,40 @@ protected static String toClassName(Type indexType, ClassInfo currentClass, Clas
...
         }
     }
 
+    private static Type resolveTypeVariable(TypeVariable typeVariable, ClassInfo currentClass, ClassInfo actualEndpointClass,
+            IndexView indexView) {
+        if (typeVariable.bounds().isEmpty()) {
+            return Type.create(DotName.createSimple(Object.class.getName()), Kind.CLASS);
+        }
+        int pos = -1;
+        for (int i = 0; i < currentClass.typeParameters().size(); ++i) {
```
Comment (gastaldi) [inline independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java:635]: Perhaps `currentClass.typeParameters()` can be extracted to a local variable here

Comment (geoand) [reply independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java]: Done

**Review by FroMage — APPROVED** [review]
That looks fine to me.

**Pull request merged**